src/simple_http.rs

```diff
@@ -184,54 +204,51 @@ impl Builder {
     }
 
     /// Set the URL of the server to the transport.
-    pub fn url<U: Into<String>>(mut self, url: U) -> Result<Self, Error> {
-        let url = url.into();
-
+    pub fn url(mut self, url: &str) -> Result<Self, Error> {
```

Comment:
Extract to a separate pure function, returning or values in a struct, and add at least some basic sanity tests.

Reply:
There are some basic sanity tests for this using the Builder. But yeah perhaps those can be more complete.